src/main/java/io/percy/appium/metadata/AndroidMetadata.java

```diff
@@ -25,21 +25,80 @@ public String deviceName() {
             return deviceName;
         }
         Object device = driver.getCapabilities().getCapability("device");
+
+        // Try desired capabilities (legacy format)
         if (device == null) {
             Map desiredCaps = (Map) driver.getCapabilities().getCapability("desired");
-            device = desiredCaps.get("deviceName");
+            if (desiredCaps != null) {
+                device = desiredCaps.get("deviceName");
+            }
+        }
+
+        // For espresso driver
+        if (device == null) {
+            device = driver.getCapabilities().getCapability("appium:deviceName");
         }
         return device.toString();
```

Comment:
If all three capabilities return null, then `device.toString()` will give Null pointer error.
Should we add a check
```suggestion
        return device != null ? device.toString() : "Unknown";
```

Reply:
Fixed the above comment